src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -45,31 +65,24 @@ class CheckpointAcceptanceService[F[_]: Concurrent](
         .toList
     }
 
-    LiftIO[F].liftIO(dao.cluster.getNodeState).flatMap { nodeState =>
-      (nodeState, checkpoint.checkpointCacheData.checkpointBlock) match {
-        case (_, None) => Sync[F].raiseError[Unit](MissingCheckpointBlockException)
-        case (NodeState.Ready, Some(cb)) =>
-          val acceptance = for {
-            _ <- syncPending(pendingAcceptanceFromOthers, cb.baseHash)
-            _ <- Sync[F].delay { logger.debug(s"[${dao.id.short}] starting accept block: ${cb.baseHash} from others") }
-            peers <- LiftIO[F].liftIO(obtainPeers)
-            _ <- resolveMissingParents(cb, peers)
-            _ <- accept(checkpoint.checkpointCacheData)
-            _ <- pendingAcceptanceFromOthers.update(_.filterNot(_ == cb.baseHash))
-          } yield ()
-
-          acceptance.recoverWith {
-            case ex: PendingAcceptance =>
-              acceptErrorHandler(ex)
-            case error =>
-              pendingAcceptanceFromOthers.update(_.filterNot(_ == cb.baseHash)) >> acceptErrorHandler(error)
-          }
+    checkpoint.checkpointCacheData.checkpointBlock match {
+      case None => Sync[F].raiseError[Unit](MissingCheckpointBlockException)
+      case Some(cb) =>
+        val acceptance = for {
+          _ <- syncPending(pendingAcceptanceFromOthers, cb.baseHash)
+          _ <- Sync[F].delay { logger.debug(s"[${dao.id.short}] starting accept block: ${cb.baseHash} from others") }
+          peers <- LiftIO[F].liftIO(obtainPeers)
```

Comment:
If you'll pass `cluster` dependency, you can use direct reference here to `peerInfo` from `cluster`